dspace-server-webapp/src/main/java/org/dspace/app/rest/ScriptRestController.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.model.ProcessRest;
+import org.dspace.app.rest.model.ScriptRest;
+import org.dspace.app.rest.model.hateoas.ProcessResource;
+import org.dspace.app.rest.repository.ScriptRestRepository;
+import org.dspace.app.rest.utils.Utils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ControllerUtils;
+import org.springframework.hateoas.ResourceSupport;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller takes care of all the requests to the system/scripts endpoint
+ */
+@RestController
+@RequestMapping("/api/" + ScriptRest.CATEGORY + "/" + ScriptRest.PLURAL_NAME)
+public class ScriptRestController {
```

Comment:
This Controller should not be mapping the full `/api/system/scripts` path....and the JavaDocs comment on this controller is misleading. As I've noted in above comments, we need to carefully name/map new Controllers.  I agree this Controller needs to exist, but it is *not* taking care of all the requests to the scripts endpoint (as both the comment says and the name of the Controller implies).

Another way of saying this is that with this Controller, we end up with *two* controllers that map the `/api/system/scripts` path.  This can be confusing to future developers in debuggging code.
1. By default the `RestResourceController` will already map the `/api/system/scripts` path as you've defined the `ScriptsRestRepository` and `ScriptsRest` classes.  This is the controller that actually will create all the base resource methods (GET, etc)
2. Now, you are creating a second controller named `ScriptRestController` that *duplicates* this same `RequestMapping`.  While this doesn't cause an error in the code, it will result in oddities, if some future developer adds a GET method to `ScriptRestController`.... as suddenly you'd have two controllers trying to implement the same method.

As we've determined in the discussion of this PR, this `ScriptRestController` serves only one purpose.  It responds to the `/api/system/scripts/{name}/processes` endpoint (specifically defining the POST method for that endpoint).

Therefore, to help future developers understand this code, I'd kindly request we do the following:
1. Rename this Controller to be something like `ScriptProcessesController`
2. Change the RequestMapping to `@RequestMapping("/api/" + ScriptRest.CATEGORY + "/" + ScriptRest.PLURAL_NAME + "/{name)/processes")`
3. Update the JavaDocs to clarify this Controller's purpose is to only add additional subresource methods (to allow connecting scripts to processes.